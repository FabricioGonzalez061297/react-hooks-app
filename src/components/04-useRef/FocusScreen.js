import React, { useRef } from 'react';

import './FocusSceen.css';
export const FocusScreen=()=>{
   
    const inputRef= useRef();

    const handleClick=()=>{
        inputRef.current.select();
    }
   
    return (
        <>
            <h1>Focus Screen</h1>
            <br></br>

            <input
            ref={inputRef}
             className="form-control"
             placeholder="su nombre"
            />
            <br></br>

            <button 
             className="btn btn-primary"
             onClick={handleClick}
            >Focus</button>

        </>
    );
};