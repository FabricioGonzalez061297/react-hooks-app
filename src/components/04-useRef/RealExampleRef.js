import React, { useState } from 'react';
import {MultipleCustomHooks} from './../03-examples/MultipleCustomHooks';

import './FocusSceen.css';

export const RealExampleRef=()=>{


    const [show,setShow] = useState(false);


    return (
        <>
            <h1>Real Example useRef</h1>
            <hr></hr>
           
            {show && <MultipleCustomHooks/> }

            <br></br>
            <button
            className="btn btn-info mt-3 txt-center"
            onClick={()=>setShow(!show)}
            >Show/Hide</button>
             

        </>
    );
}