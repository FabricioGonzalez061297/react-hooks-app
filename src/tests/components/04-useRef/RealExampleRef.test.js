import React from 'react';
import {shallow} from 'enzyme';
import {RealExampleRef} from './../../../components/04-useRef/RealExampleRef';

describe('Pruebas en RealExampleRef.test.js',()=>{

    test('debe mostrarse correcatemente', () => {
        const wrapper = shallow(<RealExampleRef/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);

    });

    test('Debe de mostrar el componente <MultipleCustomHooks/>', () => {
       
        const wrapper= shallow(<RealExampleRef/>);

        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
        
    });
    
    

});