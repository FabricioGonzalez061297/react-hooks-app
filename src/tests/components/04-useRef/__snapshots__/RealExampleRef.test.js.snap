// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RealExampleRef.test.js debe mostrarse correcatemente 1`] = `
<Fragment>
  <h1>
    Real Example useRef
  </h1>
  <hr />
  <br />
  <button
    className="btn btn-info mt-3 txt-center"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
